app/models/support_interface/provider_user_form.rb

```diff
@@ -55,6 +57,12 @@ def provider_ids
       @provider_ids.reject(&:blank?)
     end
 
+    def self.permissions_for(provider_user)
+      provider_permissions = ProviderPermissions.manage_users.where(provider_user: provider_user)
+
+      OpenStruct.new({ manage_users: provider_permissions.pluck(:provider_id) })
```

Comment:
As a matter of taste, how would you feel about making this a class-level constant so it has a name outside the method, and a plain `Struct` so it's clearly the whole definition? eg `ProviderPermissionsOptions = Struct.new(:manage_users)`. I guess the downside of plain `Struct` is that struct arguments are positional so this might get less readable when further from the definition. Maybe an `OpenStruct` as a constant?

Reply:
I like the `Struct` route as `Struct#members` gives us the valid permissions params keys for free.

Reply:
Nice, I didn't know about that!